src/toil/wdl/wdl_functions.py

```diff
@@ -348,6 +348,37 @@ def process_and_read_file(f, tempDir, fileStore, docker=False):
     return read_file(processed_file, tempDir, fileStore, docker=docker)
 
 
+def generate_stdout_file(stdout, tempDir, stderr=False, fileStore=None):
+    """
+    Create a stdout (or stderr) file from a string or bytes object.
+    Returns a file path.
+
+    :param stdout: str|bytes
+    :param tempDir: str directory
+    :param stderr: bool
+    :param fileStore:
+    """
+    if stdout is None:
+        raise RuntimeError(f'Error generating stdout file to {tempDir}')
+
+    # TODO: Not sure if this is the best way, but we probably need to differentiate the different
+    # stdout files since we move the outputs to the current working directory.
+    # Cromwell generates a folder for each task so the file is simply named stdout and lives in
+    # the task execution folder.
+
+    # the generated file would have a name like 'ten6l91dw_stdout'
+    prefix = tempDir.split('/')[-1]
+    path = os.path.join(tempDir, 'execution', '{}_{}'.format(prefix, 'stderr' if stderr else 'stdout'))
+    with open(path, 'ab+') as f:
+        f.write(stdout)
+
+    # should we import to filestore?
```

Comment:
We would need to in order to work on any kind of distributed system.  Once the workflow completes, the file would need to be exported from the filestore to obtain the final output path, wherever that may be.

Our WDL support currently only works with a local jobstore however, and this is not necessary for the current local-only functionality.  We would have to open a PR supporting AWS or Google jobstore support in WDL, in which case this would become necessary.

Reply:
>Our WDL support currently only works with a local jobstore however

What kind of crimes is our generated WDL code doing to make this happen? Is it reaching inside the job store instead of restricting itself to the FileStore API?

Reply:
> Our WDL support currently only works with a local jobstore however

I could be wrong but does this refers to how the generated WDL code writes temporary files to  `fileStore.getLocalTempDir()`? This is essentially what I’m doing here as well, since stdout and stderr files are temporary if not used ( unless we want to preserve them?)

```wdl
task task1 {
  command {
    python do_stuff.py
  }
  output {
    File results = stdout()
  }
}
```
In the case like this, we would then copy the temp stdout/stderr file to the specified output directory, along with other files defined in the output block. I can see that the output files are exported to the local jobstore here, but I suppose we could export to other jobstores as well? 

https://github.com/DataBiosphere/toil/blob/b2b09b1d912fa68052104f1bd22311d46bd03ed1/src/toil/wdl/wdl_functions.py#L263

> ... and this is not necessary for the current local-only functionality.

It'd be nice if I could get some clarification since I'm not sure if I quite understand this part.

Reply:
@adamnovak It uses the filestore API, but it has been hardcoded to use a local jobstore and the default args I believe.  I suspect it wouldn't be a lot of work to convert to running on AWS/Google, but I'd have to investigate it, and since it's never been attempted, there may or may not be blockers to doing so.  The toil args are handed down as is, so in theory it should know what to do once that's changed.  I'll try and tinker with it to see if I can get it to work.

@w-gao That sounds right to me and should work.  The temp files only need to live for the life of the worker before export.